src/Handler/TagHandler.php

```diff
@@ -40,18 +45,20 @@ class TagHandler
     /**
      * Constructor
      *
-     * @param CacheInvalidator $invalidator The invalidator instance.
-     * @param string           $tagsHeader  Header to use for tags, defaults to X-Cache-Tags.
+     * @param CacheInvalidator $invalidator  The invalidator instance.
+     * @param string           $tagsHeader   Header to use for tags, defaults to X-Cache-Tags.
+     * @param int              $headerLength Maximum header size in bytes, defaults to 7500.
      *
      * @throws UnsupportedProxyOperationException If CacheInvalidator does not support invalidate requests
      */
-    public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cache-Tags')
+    public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cache-Tags', $headerLength = 7500)
```

Comment:
ok, i didn't know that was possible…

moved the [``foreach``](https://github.com/FriendsOfSymfony/FOSHttpCache/pull/273#discussion_r51232912) outside the ``if`` statement like you suggested